lib/matplotlib/backend_bases.py

```diff
@@ -1631,7 +1667,61 @@ def __init__(self, name, canvas, key, x=0, y=0, guiEvent=None):
         self.key = key
 
 
-class FigureCanvasBase(object):
+class ExpandableBase(object):
```

Comment:
This goes on ``FigureCanvasBase`` not on ``ToolContainerBase``.  The figcanvas expands to fill all available room, the toolbar does not.  A toolbar should stay as a "bar" :wink:.

Right now I agree it looks simpler to just add this as a property to ``FigureCanvasBase``, but in the longer term I think it better and easier to start like this, and I say that for two reasons:
1. Imagine we later want to tag other (new) classes as Expandable, it becomes easier in debugging to see such a tag by their class definition, then to hunt in the code for a property... also with debugging what happens if someone accidentally adds an expandable property to the class for some other reason?  By making this a class the intention becomes clear.  By making it an attribute it could mean something else and perhaps a bit more difficult to debug...
2. Later we want to add more properties/methods around this attribute (say minimum/maximum size, or add constraints between width/height, expandable but only in one direction, or whatever), so much so we want to turn it into a class like this.  Doing so doesn't just mean we have to touch ``FigureCanvasBase`` (and other classes we might have latter set this attribute on), but also all of the backends.

For these reasons I think it best to start of with clear readable and maintainable code, even if it feels like overkill right now.